File: src/lib/aircraft-registry.js
import * as pa28161Climb from './pa28-161-climb-data.js';
import * as pa28161Cruise from './pa28-161-cruise-data.js';
import * as pa28161Engine from './pa28-161-engine-data.js';
import * as pa28181Climb from './pa28-181-climb-data.js';
import * as pa28181Cruise from './pa28-181-cruise-data.js';
import * as pa28181Engine from './pa28-181-engine-data.js';

const REGISTRY = {
    'pa28-161': { climb: pa28161Climb, cruise: pa28161Cruise, engine: pa28161Engine },
    'pa28-181': { climb: pa28181Climb, cruise: pa28181Cruise, engine: pa28181Engine },
};

export function getAircraftData(aircraftType) {
    return REGISTRY[aircraftType] ?? null;
}

File: ios/AvCalcEngineKit/Sources/AvCalcEngineKit/AircraftRegistry.swift
// Mirrors src/lib/aircraft-registry.js's REGISTRY map. This is the single indirection point
// from an aircraft-type string to its digitized data — kept in Swift (not JS) deliberately,
// since the whole point of this native module is that the digitized POH numbers never exist
// as plain-text JS in the shipped app. The JS side only ever passes an aircraftType string
// across the Capacitor bridge; Swift resolves it here.

public enum AircraftRegistry {
    public static let all: [String: AircraftData] = [
        "pa28-151": AircraftData(climb: PA28151Data.climb, cruise: PA28151Data.cruise, engine: PA28151Data.engine, airspeedCal: PA28151Data.airspeedCal),
        "pa28-161": AircraftData(climb: PA28161Data.climb, cruise: PA28161Data.cruise, engine: PA28161Data.engine, airspeedCal: PA28161Data.airspeedCal),
        "pa28-181": AircraftData(climb: PA28181Data.climb, cruise: PA28181Data.cruise, engine: PA28181Data.engine, airspeedCal: PA28181Data.airspeedCal),
    ]

    public static func data(for aircraftType: String) -> AircraftData? {
        return all[aircraftType]
    }
}
